### ISSUE TITLE
silkworm_capi_test ODR violation in ASAN build

### ISSUE THREAD

**Issue by canepat** [erigontech/silkworm#1881]
After #1879 was merged into master, an apparently new ODR violation in `silkworm_capi_test` target started to show up with two different symptoms in ASAN build:
- explicit ODR violation for `mdbx_build` symbol during runtime: see [here](https://app.circleci.com/pipelines/github/erigontech/silkworm/10929/workflows/b22c4e5e-f1eb-4701-a8a0-776b192e3e8a/jobs/48432)
- implicit ODR violation hidden behind a weird Protobuf runtime error: see [here](https://app.circleci.com/pipelines/github/erigontech/silkworm/10929/workflows/b22c4e5e-f1eb-4701-a8a0-776b192e3e8a/jobs/48431)

As a result, `silkworm_capi_test` target is currently skipped in master for ASAN build.
